addons/hr_holidays_attendance/models/hr_leave_allocation.py

```diff
@@ -62,10 +75,14 @@ def write(self, vals):
                 allocation.overtime_id.sudo().duration = -1 * duration
         return res
 
+    def unlink(self):
+        self.overtime_id.sudo().unlink()
+        return super().unlink()
```

Comment:
What is the reason for this change? It doesn't seem to be mentioned in the spec.